src/core/tsi/ssl_transport_security.cc

```diff
@@ -1254,6 +1255,57 @@ static tsi_result ssl_handshaker_result_extract_peer(
   return result;
 }
 
+static tsi_result ssl_handshaker_result_extract_local_peer(
+    const tsi_handshaker_result* self, tsi_peer* local_peer) {
+  tsi_result result = TSI_OK;
+  const unsigned char* alpn_selected = nullptr;
+  unsigned int alpn_selected_len;
+  const tsi_ssl_handshaker_result* impl =
+      reinterpret_cast<const tsi_ssl_handshaker_result*>(self);
+  X509 *local_cert = SSL_get_certificate(impl->ssl);
```

Comment:
Please check that `impl->ssl` is not nullptr before passing it into `SSL_get_certificate`. :)